### PR TITLE
Stripe prices

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

https://github.com/mysociety/whatdotheyknow-private/issues/328
Depends on #8405
Depends on #8407 

## What does this do?

Switches from Stripe Plans to Prices allowing us to change subscriptions and update Pro our pricing model.

### CODE REVIEW

**Comment by gbp** [comment]
@dracos it has been suggested that you might be the best places to review this. There are a lot of additional changes in the linked PRs which really someone should review but not sure if you've really got time to do that?

**Comment by dracos** [comment]
Sure, I can take a look at them all

**Review by dracos — COMMENTED** [review]
Looks good, couple of queries is all :)

File: app/models/alaveteli_pro/price.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+##
+# A wrapper for a Stripe::Price
+#
+class AlaveteliPro::Price < SimpleDelegator
+  include Taxable
+
+  tax :unit_amount
+
+  def self.list
+    AlaveteliConfiguration.stripe_prices.inject([]) do |arr, (_key, id)|
```
[the remark this answers is not in the document]
Comment (gbp) [reply app/models/alaveteli_pro/price.rb]: No benefit - inject was left over from when I had a different structure for `AlaveteliConfiguration.stripe_prices`. Originally I was think of having an `enabled` key and was filtering here but this just added unnecessary complexity.

File: spec/views/alaveteli_pro/plans/index.html.erb_spec.rb [hunk 2]
```diff
@@ -7,19 +7,19 @@
...
-  let(:plan) { AlaveteliPro::Plan.new(stripe_plan) }
+  let(:price) { AlaveteliPro::Price.new(stripe_price) }
   let(:cents_price) { 880 }
 
-  let(:stripe_plan) do
-    stripe_helper.create_plan(
-      id: 'pro', product: product.id, amount: cents_price
+  let(:stripe_price) do
+    stripe_helper.create_price(
+      id: 'pro', product: product.id, unit_amount: cents_price
```
[the remark this answers is not in the document]
Comment (gbp) [reply spec/views/alaveteli_pro/plans/index.html.erb_spec.rb]: I have done for this file. Other prices in plans and subscription controller tests I have left as `pro` is still the default value in `lib/configuration.rb`. This is so upgrade is easier for re-users

**Comment by gbp** [comment]
After running the final tests on the staging environment, I discovered this required some minor changes to the migration rake task. I have pushed a fixup to address the issue. Apart from this, everything appears to be in order.

**Pull request merged**